hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/repair/om/TestFSORepairTool.java

```diff
@@ -0,0 +1,576 @@
+import org.apache.hadoop.ozone.client.OzoneClientFactory;
+import org.apache.hadoop.ozone.client.io.OzoneOutputStream;
+import org.apache.hadoop.ozone.om.OMStorage;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.OmDirectoryInfo;
+import org.apache.hadoop.ozone.repair.OzoneRepair;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.hadoop.ozone.OzoneConsts.OM_DB_NAME;
+import static org.apache.hadoop.ozone.OzoneConsts.OZONE_OFS_URI_SCHEME;
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_ADDRESS_KEY;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * FSORepairTool test cases.
+ */
+public class TestFSORepairTool {
+  public static final Logger LOG = LoggerFactory.getLogger(TestFSORepairTool.class);
+  private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+  private final ByteArrayOutputStream err = new ByteArrayOutputStream();
+  private static final PrintStream OLD_OUT = System.out;
+  private static final PrintStream OLD_ERR = System.err;
+  private static final String DEFAULT_ENCODING = UTF_8.name();
+  private MiniOzoneCluster cluster;
+  private FileSystem fs;
+  private OzoneClient client;
+  private OzoneConfiguration conf = null;
+
+  @BeforeEach
+  public void init() throws Exception {
+    // Set configs.
+    conf = new OzoneConfiguration();
+
+    // Build cluster.
+    cluster =  MiniOzoneCluster.newBuilder(conf).build();
+    cluster.waitForClusterToBeReady();
+
+    // Init ofs.
+    final String rootPath = String.format("%s://%s/", OZONE_OFS_URI_SCHEME, conf.get(OZONE_OM_ADDRESS_KEY));
+    conf.set(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, rootPath);
+    fs = FileSystem.get(conf);
+    client = OzoneClientFactory.getRpcClient(conf);
+
+    System.setOut(new PrintStream(out, false, DEFAULT_ENCODING));
+    System.setErr(new PrintStream(err, false, DEFAULT_ENCODING));
+  }
+
+  @AfterEach
+  public void reset() throws IOException {
+    // reset stream after each unit test
+    out.reset();
+    err.reset();
+
+    // restore system streams
+    System.setOut(OLD_OUT);
+    System.setErr(OLD_ERR);
+
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+    if (client != null) {
+      client.close();
+    }
+    IOUtils.closeQuietly(fs);
+  }
+
+  @Test
+  public void testConnectedTreeOneBucket() throws Exception {
+    CommandLine cmd = new CommandLine(new OzoneRepair()).addSubcommand(new CommandLine(new OMRepair())
+            .addSubcommand(new FSORepairCLI()));
```

Comment:
I know this follows existing test pattern, but:

```suggestion
    CommandLine cmd = new OzoneRepair().getCmd();
```

This is simpler, and it tests the `ozone repair` command as it exists.

By using explicit `addSubcommand` in tests, the CLI could be broken and we wouldn't notice.

(This comment applies to all test cases where `new CommandLine` is created.)

(I'm making the same change in existing `TestOzoneRepairShell` in #7526.)